Sources/Build/Triple.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+/// Triple - Helper class for working with Destination.target values
+///
+/// Used for parsing values such as x86_64-apple-macosx10.10 into
+/// set of enums. For os/arch/abi based conditions in build plan.
+///
+/// @see Destination.target
```

Comment:
Might be good to have a reference here to LLVM's implementation, if this was inspired by it

Reply:
You mean just reference to file?
Some references already exist in Destination.target

Reply:
Reference to https://github.com/apple/swift-llvm/blob/stable/include/llvm/ADT/Triple.h added

Reply:
Ya thats all I meant